code/iaas/lb-instance/src/main/java/io/cattle/platform/lb/instance/dao/impl/LoadBalancerInstanceDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package io.cattle.platform.lb.instance.dao.impl;
+
+import static io.cattle.platform.core.model.tables.NetworkTable.NETWORK;
+import io.cattle.platform.core.constants.NetworkConstants;
+import io.cattle.platform.core.dao.AccountDao;
+import io.cattle.platform.core.dao.GenericMapDao;
+import io.cattle.platform.core.dao.NetworkDao;
+import io.cattle.platform.core.model.LoadBalancer;
+import io.cattle.platform.core.model.LoadBalancerHostMap;
+import io.cattle.platform.core.model.Network;
+import io.cattle.platform.db.jooq.dao.impl.AbstractJooqDao;
+import io.cattle.platform.lb.instance.dao.LoadBalancerInstanceDao;
+import io.cattle.platform.object.ObjectManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.inject.Inject;
+
+public class LoadBalancerInstanceDaoImpl extends AbstractJooqDao implements LoadBalancerInstanceDao {
+
+    @Inject
+    GenericMapDao mapDao;
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Inject
+    NetworkDao ntwkDao;
+
+    @Inject
+    AccountDao accountDao;
+
+    @Override
+    public List<Long> getLoadBalancerHosts(long lbId) {
+        List<Long> hostIds = new ArrayList<Long>();
+        for (LoadBalancerHostMap map : mapDao.findToRemove(LoadBalancerHostMap.class, LoadBalancer.class, lbId)) {
+            hostIds.add(map.getHostId());
+        }
+        return hostIds;
+    }
+
+    @Override
+    public Network getLoadBalancerInstanceNetwork(LoadBalancer loadBalancer) {
+        // today the only network in cattle is owned by the system account, so passing its id
+        // in the future, will pass the lb owner id
+        List<? extends Network> networks = ntwkDao.getNetworksForAccount(accountDao.getSystemAccount().getId(),
+                NetworkConstants.KIND_HOSTONLY);
+        if (networks.isEmpty()) {
+            throw new RuntimeException(
+                    "Load Balancer account doesn't own a network where Load Balancer Instance can be started");
+        }
+        return objectManager.findOne(Network.class, NETWORK.ID, networks.get(0).getId());
```

Comment:
You can just return networks.get(0).  There's no difference in an object loaded from a jOOQ query or from the ObjectManager.

Also, if you do want to load an object from the ObjectManager, you can use ObjectManager.loadResource(Network.class, 42L)